ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -335,8 +343,13 @@ private int findIndex(StoreKey keyToFind, ByteBuffer mmap)
    * @param fileEndOffset The file end offset that this entry represents.
    * @throws StoreException
    */
-  public void addEntry(IndexEntry entry, long fileEndOffset)
+  void addEntry(IndexEntry entry, long fileEndOffset)
       throws StoreException {
+    if (!entry.getValue().getOffset().getName().equals(getLogSegmentName())) {
+      throw new IllegalArgumentException("Log segment referred to by the index value [" +
+          entry.getValue().getOffset().getName() + "] is not the same as the one referred to by this index segment ["
+          + indexFile.getName() + "]");
```

Comment:
It's the same since the index file name has the name of the log segment within it but i have removed this check for now since I noticed there are no checks to see if key size and value size match.